onchain/rollups/contracts/consensus/quorum/Quorum.sol

```diff
@@ -3,32 +3,41 @@
 
 pragma solidity ^0.8.8;
 
-import {AccessControlEnumerable} from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
-import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 import {PaymentSplitter} from "@openzeppelin/contracts/finance/PaymentSplitter.sol";
-
 import {AbstractConsensus} from "../AbstractConsensus.sol";
+import {IConsensus} from "../IConsensus.sol";
 import {IHistory} from "../../history/IHistory.sol";
 
 /// @title Quorum consensus
 /// @notice A consensus model controlled by a small set of addresses, the validators.
-/// @dev This contract uses several OpenZeppelin contracts:
-/// `AccessControlEnumerable`, `EnumerableSet`, and `PaymentSplitter`.
-/// For more information on those, please consult OpenZeppelin's official documentation.
-contract Quorum is AbstractConsensus, AccessControlEnumerable, PaymentSplitter {
-    using EnumerableSet for EnumerableSet.AddressSet;
-
-    /// @notice The validator role.
-    /// @dev Only validators can submit claims.
-    bytes32 public constant VALIDATOR_ROLE = keccak256("VALIDATOR_ROLE");
-
+///         In this version, we assume the quorum doesn't change after the constructor.
+contract Quorum is AbstractConsensus, PaymentSplitter {
     /// @notice The history contract.
     /// @dev See the `getHistory` function.
     IHistory internal immutable history;
 
-    /// @notice For each claim, the set of validators that agree
-    /// that it should be submitted to the history contract.
-    mapping(bytes => EnumerableSet.AddressSet) internal yeas;
+    // Quorum members
+    // Map an address to true if it's a validator
+    mapping(address => bool) public validators;
```

Comment:
Yes. The main idea is to reduce `AccessControlEnumerable` to only something like `AccessControl`. But the reason why I didn't use `AccessControl` is because `AccessControl` is most useful when there are different roles, while we only have one role, `validator`. So it doesn't make sense to me to do 1 extra storage read (through [this variable](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/fd81a96f01cc42ef1c9a5399364968d0e07e9e90/contracts/access/AccessControl.sol#L56)) each time when we want to read the validators.

Reply:
This doesn't imply an extra storage read.
It just adds a new `keccak` in the calculation of the storage slot where the value resides.
Either way, I think it is still worth it to use `AccessControl` because it has been tested thoroughly.

Reply:
You are right there's no extra storage read. Sorry about my mistake.
But you see, all we need is a `mapping(address => bool)`, set some addresses map to `true` in the constructor, plus we used it for the modifier. This is simple enough that it doesn't make much of a difference using existing codes or not, since it's only 2 lines of codes plus a modifier.

Another advantage is that it can be more flexible if we want to optimize further. If the current version is ok, I'm creating another PR based on the current version for the last step of optimization. We can decide on this after the other PR, ok? If the other PR is not gonna be merged, then we can use `AccessControl` if you think it's better.

Reply:
> But you see, all we need is a `mapping(address => bool)`, set some addresses map to true in the constructor, plus we used it for the modifier.

You're right, but this logic has already been implemented. There is no need to re-implement it. Besides, for anyone reading this code, they will need to understand what you're trying to achieve with this mapping, the modifier, etc. But if you use `AccessControl` (which many smart contract developers are already familiar with), the code will be much clearer, and, most importantly, more trustworthy.